modules/oe_content_event/src/Plugin/Validation/Constraint/EventFieldsRequiredValidator.php

```diff
@@ -91,21 +91,37 @@ protected function validateGroupFieldsEmpty(array $fields, Constraint $constrain
    *   The node object.
    */
   protected function validateOrganiserGroupFields(Constraint $constraint, NodeInterface $node) {
-    $violation = NULL;
     if ($node->get('oe_event_organiser_internal')->isEmpty() && $node->get('oe_event_organiser_name')->isEmpty()) {
-      $violation = $this->context->buildViolation('You have to fill in at least one of the following fields: @internal or @organiser_name', [
+      return;
+    }
+
+    $violation = NULL;
+    $oe_event_organiser_is_internal_value = (bool) $node->get('oe_event_organiser_is_internal')->value;
+
+    if (!$oe_event_organiser_is_internal_value && !$node->get('oe_event_organiser_internal')->isEmpty()) {
+      // If checkbox "Organiser is internal" isn't checked,
+      // field "Internal organiser" has to be empty.
+      $violation = $this->context->buildViolation('When @is_internal field is not checked, @internal field have to be empty.', [
+        '@is_internal' => $node->getFieldDefinition('oe_event_organiser_is_internal')->getLabel(),
         '@internal' => $node->getFieldDefinition('oe_event_organiser_internal')->getLabel(),
+      ]);
+    }
```

Comment:
Why was this implemented? Was not part of the ticket and there are no such requirements. This we do automatically in oe_content_event_node_presave() and should not throw validation error.

Reply:
Please see the ticket implementing that OPENEUROPA-2550 and it is written: 
![image](https://user-images.githubusercontent.com/22004498/91312165-39ea3000-e7b4-11ea-9296-c547d8e4cb57.png)

Reply:
@ademarco please discuss with @nagyad correct approach.

Reply:
Ye, @nagyad is right. The case is the following: if a malicious user fills in both name and reference, this code will give an error, while PO doesn't want that, he only wants the case to be taken care silently, which we already do in the presave. The right thing to do here is to just remove the validation.